jerry-core/ecma/builtin-objects/ecma-builtin-function-prototype.c

```diff
@@ -210,11 +211,37 @@ ecma_builtin_function_prototype_object_bind (ecma_object_t *this_arg_obj_p , /**
   /* 4. 11. 18. */
   ecma_object_t *prototype_obj_p;
 
+#if ENABLED (JERRY_ES2015_BUILTIN_PROXY)
```

Comment:
This code part has a bit too much marcos and bit hard to read (macro guarded {}-s). Would it make sense to move this to it's own function?
like:
```c
ecma_object_t *bind_get_proto (ecma_object_t *this_arg_obj_p)
{
#if ENABLED (JERRY_ES2015_BUILTIN_PROXY)
  if (ecma_object_is_proxy (this_arg_obj_p))
  {
    ecma_value_t proto = ecma_proxy_object_get_prototype_of (this_arg_obj_p);

    if (ECMA_IS_VALUE_ERROR (proto))
    {
      return proto;
    }
    if (ecma_is_value_null (proto))
    {
      return NULL;
    }
    return ecma_get_object_from_value (proto);
  }
#endif /* ENABLED (JERRY_ES2015_BUILTIN_PROXY) */


#if ENABLED (JERRY_ES2015)
  ecma_object_t *prototype_obj_p = NULL;
  jmem_cpointer_t proto_cp = ecma_op_ordinary_object_get_prototype_of (this_arg_obj_p);
  if (proto_cp != JMEM_CP_NULL)
  {
    prototype_obj_p = ECMA_GET_NON_NULL_POINTER (ecma_object_t, proto_cp);
    ecma_ref_object (prototype_obj_p);
  }
  return prototype_obj_p;
#else /* !ENABLED (JERRY_ES2015) */
  return ecma_builtin_get (ECMA_BUILTIN_ID_FUNCTION_PROTOTYPE);
#endif /* ENABLED (JERRY_ES2015) */
}
```

Reply:
I'm not sure about that makes this codepath more readable. Also it the Proxy.[[GetPrototypeOf]] can also throw an error therefore the

```
if (ECMA_IS_VALUE_ERROR (proto))
    {
      return proto;
    }
```
return is incorrect, since its an ecma-value. We can do a workaround by returning with 0x01 when error happens, but this returned value also should be check in the caller site. So IMO separating this part into an other function does not worth it. However the code can slightly be simplified, I'll update soon.